framework/src/main/java/org/checkerframework/framework/flow/CFAbstractTransfer.java

```diff
@@ -366,6 +376,57 @@ public S initialStore(UnderlyingAST underlyingAST, List<LocalVariableNode> param
     return store;
   }
 
+  /**
+   * Determine whether a lambda expression is leaked.
+   *
+   * <p>This currently checks for the {@link NonLeaked} annotation, which is trusted.
+   *
+   * @param lambda the lambda expression
+   * @param aTypeFactory an annotated type factory
+   * @return true if the lambda expression is found to be annotated with {@link NonLeaked}
+   */
+  private boolean isLambdaLeaked(CFGLambda lambda, AnnotatedTypeFactory aTypeFactory) {
+    LambdaExpressionTree lambdaTree = lambda.getLambdaTree();
+    TreePath lambdaPath = aTypeFactory.getPath(lambdaTree);
+    Tree lambdaParent = lambdaPath.getParentPath().getLeaf();
+    if (lambdaParent.getKind() == Tree.Kind.METHOD_INVOCATION) {
+      MethodInvocationTree invok = (MethodInvocationTree) lambdaParent;
+      ExecutableElement methodElt = TreeUtils.elementFromUse(invok);
+      AliasingAnnotatedTypeFactory aliasingAtf =
+          analysis
+              .atypeFactory
+              .getChecker()
+              .getTypeFactoryOfSubcheckerOrNull(AliasingChecker.class);
+      if (aliasingAtf != null) {
+        // Check whether any parameter is NOT annotated with @NonLeaked
+        return methodElt.getParameters().stream()
+            .anyMatch(
+                param ->
+                    aliasingAtf.getAnnotatedType(param).getPrimaryAnnotation(NonLeaked.class)
+                        == null);
+      }
+    }
+    return true;
+  }
+
+  private boolean isLambdaPure(CFGLambda lambda, AnnotatedTypeFactory aTypeFactory) {
+    LambdaExpressionTree lambdaTree = lambda.getLambdaTree();
+    if (lambdaTree.getBodyKind() == LambdaExpressionTree.BodyKind.EXPRESSION) {
+      ExpressionTree lambdaExpression = (ExpressionTree) lambdaTree.getBody();
+      JavaExpression internalRepr = JavaExpression.fromTree(lambdaExpression);
+      List<Element> methodsInvoked = JavaExpression.methodsFromMethodCall(internalRepr);
+      List<JavaExpression> methodArguments = JavaExpression.argumentsFromMethodCall(internalRepr);
+      boolean areAllArgumentsUnassignable =
+          methodArguments.stream().allMatch(Predicate.not(JavaExpression::isAssignableByOtherCode));
+      boolean isMethodCallSequencePure =
+          methodsInvoked.stream()
+              .allMatch(
+                  methodInvok -> aTypeFactory.getDeclAnnotation(methodInvok, Pure.class) != null);
```

Comment:
@mernst This works for methods that are explicitly annotated with `@Pure`, but won't work for other cases, such as when the CF is run with `-AassumePure`.

Is there a way to obtain the value of that flag from this location?

Reply:
Can you run `checker.hasOption("assumePure")`?